src/tcgwars/logic/impl/gen7/LostThunder.groovy

```diff
@@ -2406,17 +2406,26 @@ public enum LostThunder implements LogicCardInfo {
           resistance FIGHTING, MINUS20
           bwAbility "Mirror Counter" , {
             text "If this Pokémon is your Active Pokémon and is damaged by an attack from your opponent's Pokémon-GX or Pokémon-EX (even if this Pokémon is Knocked Out), put damage counters on the Attacking Pokémon equal to the damage done to this Pokémon."
+
+
             delayedA (priority:LAST) {
+              def counterDmg = 0
               before APPLY_ATTACK_DAMAGES, {
                 if(self.active && ef.attacker.owner != self.owner && (ef.attacker.pokemonGX || ef.attacker.pokemonEX)) {
                   bg.dm().each{
                     if(it.to == self && it.dmg.value) {
-                      bc "Mirror Counter countered ${ef.attacker}'s attack"
-                      directDamage(it.dmg.value, ef.attacker, Source.SRC_ABILITY)
+                      counterDmg = it.dmg.value
                     }
                   }
                 }
               }
+              after APPLY_ATTACK_DAMAGES, {
+                if (counterDmg) {
+                  bc "Mirror Counter countered ${ef.attacker}'s attack"
+                  directDamage(counterDmg, ef.attacker, Source.SRC_ABILITY)
+                  counterDmg = 0
+                }
+              }
```

Comment:
3
Might as well use the static here too.

**However** I'd actually recommend adding another closure to the static that gets called before applyEffect gets set to True so you can affect both parts of the static freely.
```suggestion
              ifActiveAndDamagedByAttackAttached {
                if (counterDmg) {
                  bc "Mirror Counter countered ${ef.attacker}'s attack"
                  directDamage(counterDmg, ef.attacker, Source.SRC_ABILITY)
                  counterDmg = 0
                }
              }
```